homeassistant/components/neato/camera.py

```diff
@@ -102,33 +111,34 @@ def update(self):
 
         self._image = image.read()
         self._image_url = image_url
-        self._generated_at = map_data["generated_at"]
+        if data := map_data["generated_at"]:
+            self._generated_at = data
```

Comment:
Personally, I find that the walrus operator does not improve readability. Also, list access should not be more time consuming than variable access, so we don't gain anything from that either.

Reply:
As far as I know, it's the preferred approach if the if doesn't spawn on multi lines.

Reply:
Don't you think it's easier to read without the additional variable and a rearly used operator?

Reply:
Or even better:

`self._generated_at = map_data.get("generated_at", None)`

Reply:
Indeed, how can it be I missed this... code updated.